storage/Devices/MdImpl.cc

```diff
@@ -915,6 +919,14 @@ namespace storage
 		break;
 	}
 
+	// adjust block size
+	if (block_size && block_size != get_region().get_block_size())
+	{
+	    Region region = Region(get_region());
```

Comment:
why not simple `Region region(get_region());` ?

Reply:
That would also do; I was probably just in a typing mood.